src/storage/src/metrics/upsert.rs

```diff
@@ -95,12 +91,6 @@ impl UpsertMetricDefs {
                     this source",
                 var_labels: ["source_id", "worker_id"],
             )),
-            rocksdb_autospill_in_use: registry.register(metric!(
-                name: "mz_storage_upsert_state_rocksdb_autospill_in_use",
-                help: "Flag to denote whether upsert state has spilled to rocksdb \
-                    or using in-memory state",
-                var_labels: ["source_id", "worker_id"],
-            )),
```

Comment:
Perhaps unexpectedly, this metric is only used with the autospill backend. If the in-memory or the rocksdb backend is used, no metric values are written.

We could change the metric to track use of the in-memory/rocksdb backend, but given that we plan to remove the former one as a follow-up, that seems like a waste of time.